### PR TITLE
[DOCS] Add rollup V2 API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Documents new `<index>/_rollup` API
* Updates rollup APIs overview page
* Enables rollups V2 in docs integ tests

Relates to #64900

### Scope
This PR is focused on API reference documentation This PR does not include a more general overview or tutorial for rollups V2. I plan to cover those in later PRs. I also plan to open additional PR(s) to better label legacy rollup APIs.

### Previews
* Rollup APIs overview: https://elasticsearch_65398.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/rollup-apis.html
* Rollup API docs: https://elasticsearch_65398.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/rollup-api.html

### CODE REVIEW

File: docs/reference/rollup/apis/rollup-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+later. For example, you can't aggregate daily rollups into hourly
+summaries. However, smaller time intervals can greatly increase the size of your
+`rollup_index`.
+
+`time_zone`::
+(Optional, string)
+Time zone for the `field`. Valid values are ISO 8601 UTC offsets, such as
+`+01:00` or `-08:00`, and IANA time zone IDs, such as `America/Los_Angeles`.
+Defaults to `+00:00` (UTC).
+=====
```
Comment (jrodewig) [inline docs/reference/rollup/apis/rollup-api.asciidoc:117]: `delay` is also a valid param for `date_histogram`, but I chose not to include it as it seems to be no-op for the API endpoint. Let me know if that's not the case.

File: docs/reference/rollup/apis/rollup-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+[[rollup-api-request]]
+==== {api-request-title}
+
+`PUT /<index>/_rollup`
+
+[[rollup-api-prereqs]]
+==== {api-prereq-title}
+
+If the {es} {security-features} are enabled, you must have the `manage` cluster
+privilege to use this API. See <<security-privileges>>.
```
Comment (jrodewig) [inline docs/reference/rollup/apis/rollup-api.asciidoc:52]: Note for reviewers: The `manage_rollup` privilege doesn't seem to have any affect on this API. If this is intentional, I'll update the privileges docs in another PR.

File: docs/reference/rollup/rollup-apis.asciidoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+another index, called a rollup index. For example, you can aggregate hourly data
+into daily or weekly summaries.
+
+* <<rollup-api,Rollup API>>
+
+[discrete]
+[[legacy-rollup-apis]]
+=== Legacy rollup APIs
+
+Prior to {es} 7.x, you could only create a rollup using a periodic cron jobs.
```
Comment (jrodewig) [inline docs/reference/rollup/rollup-apis.asciidoc:18]: > Prior to {es} 7.x, you could only create a rollup using a periodic cron jobs.

I'll update 7.x with the actual release version once we're closer to that.

**Comment by jrodewig** [comment]
@talevy This is ready for a review at your convenience. I left a few questions and comments in the file changes. Thanks!

File: docs/build.gradle [hunk 4]
```diff
@@ -52,6 +52,9 @@ testClusters.integTest {
   if (singleNode().testDistribution == DEFAULT) {
     setting 'xpack.license.self_generated.type', 'trial'
     setting 'indices.lifecycle.history_index_enabled', 'false'
+    if (BuildParams.isSnapshotBuild() == true) {
```
Comment (talevy) [inline docs/build.gradle:55]: now that I am seeing this, I may have to change how I've done this in my PRs.

I feel it is safe to always set the system property in test clusters. I'll change all of them at a later point

File: docs/reference/rollup/apis/rollup-api.asciidoc (hunk 2 above)
Comment (talevy) [reply docs/reference/rollup/apis/rollup-api.asciidoc:52]: I think you found a bug :)

File: docs/reference/rollup/apis/rollup-api.asciidoc (hunk 1 above)
Comment (talevy) [reply docs/reference/rollup/apis/rollup-api.asciidoc:113]: This is one of those WIP leaks. the config is being re-used, but you're right, the `delay` option is not intended to be used in this action, so it should not be mentioned

File: docs/reference/rollup/rollup-apis.asciidoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+another index, called a rollup index. For example, you can roll up hourly data
+into daily or weekly summaries.
+
+* <<rollup-api,Rollup API>>
+
+[discrete]
+[[legacy-rollup-apis]]
+=== Legacy rollup APIs
+
+Prior to {es} 7.x, you could only create a rollup using a periodic cron jobs.
```
Comment (talevy) [inline docs/reference/rollup/rollup-apis.asciidoc:18]: drop the "a"?

**Review by talevy — APPROVED** [review]
woohoo! thank you for this. I will follow-up on the security role issue.

File: docs/reference/rollup/rollup-apis.asciidoc (hunk 5 above)
Comment (jrodewig) [reply docs/reference/rollup/rollup-apis.asciidoc:18]: Good catch! Thanks!

File: docs/reference/rollup/apis/rollup-api.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/rollup/apis/rollup-api.asciidoc:52]: I'll open up an issue for that. Thanks, Tal!

Comment (talevy) [reply docs/reference/rollup/apis/rollup-api.asciidoc:52]: opened #65469. should fix this

Comment (jrodewig) [reply docs/reference/rollup/apis/rollup-api.asciidoc:52]: Awesome! Saved me a step. Thanks.

**Pull request merged**